src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1457,14 +1469,36 @@ static bool processSubscriptions
 
     /* Send notification */
     LM_T(LmtSubCache, ("NOT ignored: %s", trigs->cacheSubId.c_str()));
-    if (processOnChangeConditionForUpdateContext(notifyCerP,
-                                                 trigs->attrL,
-                                                 mapSubId,
-                                                 trigs->reference,
-                                                 trigs->notifyFormat,
-                                                 tenant,
-                                                 xauthToken,
-                                                 fiwareCorrelator))
+
+    CachedSubscription*  cacheP = subCacheItemLookup(tenant.c_str(), trigs->cacheSubId.c_str());
+    bool                 notificationSent;
+
+    if (cacheP)
```

Comment:
In order to avoid duplication, I'd suggest:

```
std::vector<std::string> attrV;
if (cacheP)
{
  attrV = cacheP->attributes;
}
else
{
  attrV = trigs->attrL.attributeV;
}

notificationSent = processOnChangeConditionForUpdateContext(notifyCerP,
                                                                  trigs->attrL,
                                                                  mapSubId,
                                                                  trigs->reference,
                                                                  trigs->renderFormat,
                                                                  tenant,
                                                                  xauthToken,
                                                                  fiwareCorrelator,
                                                                  attrV);
```

(Above fragment probably don't work due to typing, but I guess you get the idea...)